hw/xbox/xid-sdl.c

```diff
@@ -464,6 +479,66 @@ static void usb_xid_class_initfn(ObjectClass *klass, void *data)
     uc->handle_attach  = usb_desc_attach;
 }
 
+static HapticEmulationMode select_haptic_emulation_mode(SDL_Haptic *haptic)
+{
+    uint query_result = SDL_HapticQuery(haptic);
+    if (query_result & SDL_HAPTIC_LEFTRIGHT) {
+        return EMU_HAPTIC_LEFT_RIGHT;
+    }
+
+    printf("SDL_HapticQuery result: 0x%x\n", query_result);
```

Comment:
nit: As this is a debug feature (although meant for users) we should either use a debug printing macro, or, at least describe the issue: `"No supported haptic mode found (SDL_HapticQuery result: 0x%x). Haptic disabled."` (or something like this)

Reply:
I'll stick with your second suggestion here, as this is a user facing problem.